kobo/apps/hook/tests/test_ssrf.py

```diff
@@ -34,9 +35,10 @@ def test_send_with_ssrf_options(self):
                       content_type='application/json')
 
         # Try to send data to external endpoint
-        success = service_definition.send()
-        self.assertFalse(success)
+        with pytest.raises(SSRFProtectException):
+            service_definition.send()
```

Comment:
one idea would be to rewrite `MockSSRFProtect._get_ip_address()` so that it takes an IP address a parameter. then, you could use `functools.partial` to do this:
```python
    @patch('ssrf_protect.ssrf_protect.SSRFProtect._get_ip_address',
           new=functools.partial(MockSSRFProtect._get_ip_address, '1.2.3.4'))
```

Reply:
For the records, I did not use `functools.partial` but `MagicMock` instead. 

```python
     from ipaddress import ip_address
     from mock import MagicMock 
     
    @patch(
        'ssrf_protect.ssrf_protect.SSRFProtect._get_ip_address',
        new=MagicMock(return_value=ip_address('1.2.3.4'))
    )
    def test...
```